apps/web/src/components/ui/button-with-disable-tooltip.tsx

```diff
@@ -0,0 +1,44 @@
+import { cn } from "@/lib/utils";
+import { Button } from "./button";
+import type { ButtonProps } from "./button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "./tooltip";
+
+interface ButtonWithDisableTooltipProps extends ButtonProps {
+  tooltip: string;
+}
+
+export const ButtonWithDisableTooltip = ({
+  tooltip,
+  disabled,
+  ...props
+}: ButtonWithDisableTooltipProps) => {
+  const ButtonComponent = (
+    <Button
+      {...props}
+      disabled={disabled}
+      className={cn(disabled && "pointer-events-none")}
```

Comment:
We'll need to deconstruct the props a bit more.

```suggestion
      className={cn(className, disabled && "pointer-events-none")}
```